Update TiKV store status information (#6949)
pingcap/docs: tidb-scheduling.md

    * Whether the store is overloaded
    * Labels (See [Perception of Topology](/schedule-replicas-by-topology-labels.md))

> You can use PD control to check the status information of a TiKV store, which is divided into Up, Disconnect, Offline, Down, and Tombstone. The relationship between each status is as follows:

Author: > 下面更多是在分别解释每个状态的含义，并不是明确的 the relationship between each status。

好哒，下面除了 UP 之外每个状态同时包含了两个内容：对各状态的描述，在哪些情况下能从一个状态切换至另一个状态。另外其他文章中也把 the relationship between each status 链到这里了，所以我修改了在上面描述，同时反应这两句，PTAL， 感谢！

> You can use PD control to check the status information of a TiKV store, which is divided into Up, Disconnect, Offline, Down, and Tombstone. The relationship between each status is as follows:

Reviewer: > The following is the description of each status and the relationship between each of them

OK, 没问题，最后那句可以调整下表达：The following is a description of all statuses and their relationship.

> You can use PD control to check the status information of a TiKV store, which is divided into Up, Disconnect, Offline, Down, and Tombstone. The relationship between each status is as follows:

Author: > > The following is the description of each status and the relationship between each of them
> 
> OK, 没问题，最后那句可以调整下表达：The following is a description of all statuses and their relationship.

谢谢，已修改！

> + **Up**: The current TiKV store is in service.
> + **Disconnect**: Heartbeat messages between the PD and the TiKV store are lost for more than 20 seconds. If the lost period exceeds the time specified by `max-store-down-time`, the status changes to "Down".

Author: ```suggestion
+ **Up**: The TiKV store is in service.
+ **Disconnect**: Heartbeat messages between the PD and the TiKV store are lost for more than 20 seconds. If the lost period exceeds the time specified by `max-store-down-time`, the status changes to "Down".
```

> + **Down**: When the time that the TiKV store lost connection with the cluster has exceeded the time specified by `max-store-down-time` (30 minutes by default), the store changes to this status. This status indicates that the corresponding store status changes to Down and starts replenishing peers of each Region on the surviving store.

Author: ```suggestion
+ **Down**: When the time that the TiKV store lost connection with the cluster is longer than the time specified by `max-store-down-time` (30 minutes by default), the store changes to "Down". In this status, the store starts replenishing peers of each Region on the surviving store.
```

> + **Offline**: A TiKV store is manually taken offline through PD Control. In this status, the TiKV store is intermediately offline and performs leader transfer and Region balance. When the `leader_count/region_count` (obtained through PD Control) shows completion of leader transfer and Region balance, the store status changes to "Tombstone" from "Offline". In the "Offline" status, **do not** disable the store service and the physical server where the store is located.

Reviewer 2: `leader transfer and Region balance` 的说法不准确，感觉中文原文也得改一下，我一会在中文文档那边提个 PR 吧。事实上通过非部署工具触发下线的时候，既不会发生 leader transfer 也不会发生 region balance，准确的说法是“会开始将下线节点上面的 Region 搬离下线 TiKV Store”，而且是有条件限制的。Region balance 在 TiKV 集群里有其他的概念，容易产生歧义。

> + **Offline**: A TiKV store is manually taken offline through PD Control. In this status, the TiKV store is intermediately offline and performs leader transfer and Region balance. When the `leader_count/region_count` (obtained through PD Control) shows completion of leader transfer and Region balance, the store status changes to "Tombstone" from "Offline". In the "Offline" status, **do not** disable the store service and the physical server where the store is located.

Author: @JmPotato @shichun-0415 this part is revised, PTAL, thank you!

> + **Offline**: A TiKV store is manually taken offline through PD Control. In this status, the TiKV store is intermediately offline and performs leader transfer and Region balance. When the `leader_count/region_count` (obtained through PD Control) shows completion of leader transfer and Region balance, the store status changes to "Tombstone" from "Offline". In the "Offline" status, **do not** disable the store service and the physical server where the store is located.

Reviewer 2: It should be `or` here.

> do not disable the store service **_or_** the physical server where the store is located.

The rest LGTM.

> + **Offline**: A TiKV store is manually taken offline through PD Control. In this status, the TiKV store is intermediately offline and performs leader transfer and Region balance. When the `leader_count/region_count` (obtained through PD Control) shows completion of leader transfer and Region balance, the store status changes to "Tombstone" from "Offline". In the "Offline" status, **do not** disable the store service and the physical server where the store is located.

Reviewer: Suggestion: The store in this status moves all its Regions to other "Up" stores that meet the relocation conditions.
> The store in this status moves all Regions from this store to other stores that meet the relocation conditions and are in the "Up" status at the same time.

Suggestion: During the process that the store goes offline, if the cluster does not have target stores to relocate the Regions (for example, inadequate stores to hold replicas in the cluster), the store is always in the "Offline" status.
> During the process that the store goes offline, if the cluster does not have other target stores that meet the relocation conditions (for example, there are not enough stores that can meet the requirements for the number of replicas in the cluster anymore), the store always is in the "Offline" status.

> + **Offline**: A TiKV store is manually taken offline through PD Control. This is only an intermediate status for the store to go offline. The store in this status moves all its Regions to other "Up" stores that meet the relocation conditions. When `leader_count` and `region_count` (obtained through PD Control) both show `0`, the store status changes to "Tombstone" from "Offline". In the "Offline" status, **do not** disable the store service or the physical server where the store is located. During the process that the store goes offline, if the cluster does not have other target stores that meet the relocation conditions (for example, there are not enough stores that can meet the requirements for the number of replicas in the cluster anymore), the store always is in the "Offline" status.

Author: @JmPotato @shichun-0415 all comments are addressed, PTAL, thank you!

Reviewer: if the cluster does not have other target stores that meet the relocation conditions (for example, there are not enough stores that can meet the requirements for the number of replicas in the cluster anymore), the store always is in the "Offline" status.---**Seems this sentence is still the original one.**

> + **Disconnect**: When the heartbeat information of PD and the TiKV store is lost for more than 20 seconds, the store status changes to "Disconnect". When the lost time is longer than the time specified by `max-store-down-time`, the store status changes to "Down".
> + **Down**: When the time that the TiKV store lost connection with the cluster is longer than the time specified by `max-store-down-time` (30 minutes by default), the store changes to "Down". In this status, the store starts replenishing peers of each Region on the surviving store.
> + **Offline**: If you manually take a TiKV store offline through PD Control, the store status changes to "Offline". This is only an intermediate status that the store is taking offline. The store in this status performs the operations of leader transfer and Region balance. When the `leader_count/region_count` (obtained through PD Control) shows that both operations of leader transfer and Region balance have been completed, the store status changes to "Tombstone" from "Offline". In the "Offline" status, you should **not** disable the store service and the physical server where the store is located.
> + **Tombstone**: This status indicates that the TiKV store is completely offline. You can use `remove-tombstone` interface to safely clean up TiKV in this status.

Reviewer: ```suggestion
+ **Disconnect**: Heartbeat messages between the PD and the TiKV store are lost for more than 20 seconds. If the lost period exceeds the time specified by `max-store-down-time`, the status changes to "Down".
+ **Down**: Heartbeat messages between the PD and the TiKV store are lost for a time longer than `max-store-down-time` (30 minutes by default). In this status, the TiKV store starts replenishing replicas of each Region on the surviving store.
+ **Offline**: A TiKV store is manually taken offline through PD Control. In this status, the TiKV store is intermediately offline and performs leader transfer and Region balance. When the `leader_count/region_count` (obtained through PD Control) shows completion of leader transfer and Region balance, the store status changes to "Tombstone" from "Offline". In the "Offline" status, **do not** disable the store service and the physical server where the store is located.
+ **Tombstone**: The TiKV store is completely offline. You can use the `remove-tombstone` interface to safely clean up TiKVs in this status.
```
